### PR TITLE
KAFKA-6452: Add documentation for delegation token authentication

### PR DESCRIPTION
**Pull request by omkreddy**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by omkreddy** [comment]
@junrao  Added initial docs for delegation tokens. Pls take a look.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/321/

**Comment by omkreddy** [comment]
@junrao  pinging for review

File: docs/security.html [hunk 1]
```diff
@@ -661,6 +661,108 @@ <h3><a id="security_sasl" href="#security_sasl">7.3 Authentication using SASL</a
             old mechanism from JAAS config file. Incrementally bounce the cluster again.</li>
         </ol>
     </li>
+
+    <li><h4><a id="security_delegation_token" href="#security_delegation_token">Authentication using Delegation Tokens</a></h4>
+        <p>Delegation token based authentication is a lightweight authentication mechanism to complement existing SASL/SSL
+            methods. Delegation tokens are shared secrets between kafka brokers and clients. Delegation tokens will help processing
+            frameworks to distribute the workload to available workers in a secure environment without the added cost of distributing
+            Kerberos TGT or keytabs. See <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-48+Delegation+token+support+for+Kafka">KIP-48</a>
```
Comment (junrao) [inline docs/security.html]: This also saves the cost of distributing keystores when 2-way SSL is used.

**Review by junrao — COMMENTED** [review]
@omkreddy : Thanks for the patch. Looks good. Just a minor comment below.

File: docs/security.html (hunk 1 above)
Comment (omkreddy) [reply docs/security.html]: Thanks for the review. Updated the PR.

**Comment by junrao** [comment]
@omkreddy : Thanks for the patch. LGTM

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-test-coverage/334/

**Pull request closed without merging**